{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d26d7afd",
   "metadata": {},
   "outputs": [],
   "source": [
    "from azure.identity import DefaultAzureCredential\n",
    "from azure.keyvault.secrets import SecretClient\n",
    "import json\n",
    "import logging\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b87ef24",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Configure logging\n",
    "logging.basicConfig(level=logging.DEBUG)\n",
    "logger = logging.getLogger(__name__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d1e5e47d",
   "metadata": {},
   "outputs": [],
   "source": [
    "KEYVAULT_URI = 'https://kvfactanalysis.vault.azure.net/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "265ee39c",
   "metadata": {},
   "outputs": [],
   "source": [
    "secret_names = {\n",
    "    \"type\": \"FB-TYPE\",\n",
    "    \"project_id\": \"FB-PROJETCID\",\n",
    "    \"private_key_id\": \"FB-PRIVATEKEYID\",\n",
    "    \"private_key\": \"FB-PRIVATEKEY\",\n",
    "    \"client_email\": \"FB-CLIENTEMAIL\",\n",
    "    \"client_id\": \"FB-CLIENTID\",\n",
    "    \"auth_uri\": \"FB-AUTHURI\",\n",
    "    \"token_uri\": \"FB-TOKENURI\",\n",
    "    \"auth_provider_x509_cert_url\": \"FB-AUTHPROVIDERCERTURL\",\n",
    "    \"client_x509_cert_url\": \"FB-CLIENTURL\",\n",
    "    \"universe_domain\": \"FB-UNIVERSEDOMAIN\"\n",
    "}\n",
    "\n",
    "firebase_json = {\n",
    "    \"type\": \"service_account\",\n",
    "    \"project_id\": \"fact-check257\",\n",
    "    \"private_key_id\": \"5800407e8b1f6d04368c2e935d20e2659dd9e4d0\",\n",
    "    \"private_key\": \"\"\"-----BEGIN PRIVATE KEY-----MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCkqbvKaTaSNCqLKm8EbLN1jedYSojQP+j5gA0sqwSL7mF9fr4A5CrktbQinvubpwS6lJL/e82pWbNQHVPexycj4zei0hR/0yOZrOGGH6OpPT8bTq+vq730bSKSDNHOUzHFgyAVRVQo+kYKCLmtpHuXomw3yEDAQRKPxJdLf7kiZQS9pwAOGpjkD3P+teVp9ml/3agVWKtNcp3AMM/BaUrRfCi3rMOph4PdrG6XVAftyP9geBZUDZ0IgBe213UQ4MONLu3b/+JrFc0c6kbZdY6ysNyUjet2N+F6UFwViQE9IcDISOYtJn8LjuTQeAwTXNuLF4OlHHvV9vP5DxkRAgQpAgMBAAECggEAONc6swmUw+1sDjQstItNe7Pre0fnaHPXWWFBqvyyChfU4XxB3I7m26J5t0XINxf1gu7Mu5KiEOadBnfJPuricPn7d6IrX2CkNmU+SI1dcoto4GVOMkcHwHt35pC0UBPZ54N3LMCM58RVLxuJ9Q4Fd/eigbMH3xXXcXd8fpxtbRiBXSuuk9Vouhf7lnwQLWyO663QrvHxTylTtmWELScyIBxvDskSsC8f18IVUJAT5pMORNmPUMII8gTsCZa6gU4HoaMGk2AHK0sLiLK2xRHOSOS8KHdape+UIQ2Cx79RK9mevzi18rp/XIGEPKuLx46EgmQuZG/Sm8jvZjl4hBQgfQKBgQDl6vYz+sCfZI7jRUgtSEvjnbuzxeKXVUYGlj0rXh3VWzsXSNCFYLzESpOeEOtRiBtBuHAs9CyuoYDxBNG3KhMgrKaYyiCOi4Kt2HANHSiqDFl1eXkYySz+CuoWjNijK4iUy7dD78exeqSa4A0e/3tG7YOB+NP+cnR+yAYotY32uwKBgQC3V7VtoIpSOHEIPN3uhYg1j212G093GH8eb/6ifZPSK3YAhktK15QO5oev5NjkrAHxwYQreoBDQoEdI5SPFlXbpDj7kVP6LgOjZSHjX/qFRXRNkyYUFCOogYsBRf3nsOf6/Y87nFb7kLOaFnEN4Z+y0ZnWMLMYtyaU7k5xtu9sawKBgHimpxI52NEL/OuIlMqmcKR+RlydUD3n0rpsJqdWWUN3/3dFj58eV5COLg65hDMhwGPpfHwqmjP9rmNMCx0aKDVyRG5TS2hiD25Y8NwA3DD4EA3Uc9as4z5gzdyNhrWBh8fFa2NsNZeN60kWn+wPV0quIt+WivCwcS+CmbDNLTU3AoGAGUXj4orAxni1TBdn2Ftb3VY6Fo/JMneT/lQd27RHzi7jg0r/0f9Jj4xDpAF9zHfRhMvSNlQanirqtWOUNO01RKOsnzD3TINPSRTFyRwP4pMiZxcjppzoFXBWAwTs7SU7tSOCvQreCqt1toXla65wXyv5cscfyQZmMR1wfshi1KUCgYApqRkuP8dx4g/rgSRDOKbw+kBknUkxVgcStrSA8U+DxMiGLgpELPSui9fORb4UNloUBFalQ7VY3EpnlISoS+l/10eZVkmEUIOFuU0gcsaNCibzMWsNh4pQqsAzujz609BevvBxEl6+1Gc8jG48h8fBq3/rwMMdXcw+7PhELF0/RA==-----END PRIVATE KEY-----\"\"\",\n",
    "    \"client_email\": \"firebase-adminsdk-fbsvc@fact-check257.iam.gserviceaccount.com\",\n",
    "    \"client_id\": \"101709437156448516768\",\n",
    "    \"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\n",
    "    \"token_uri\": \"https://oauth2.googleapis.com/token\",\n",
    "    \"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\n",
    "    \"client_x509_cert_url\": \"https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-fbsvc%40fact-check257.iam.gserviceaccount.com\",\n",
    "    \"universe_domain\": \"googleapis.com\"\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "fcf6249b",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def set_keyvault_secrets():\n",
    "    try:\n",
    "        # Initialize Key Vault client\n",
    "        credential = DefaultAzureCredential()\n",
    "        client = SecretClient(vault_url=KEYVAULT_URI, credential=credential)\n",
    "\n",
    "        # Set each secret in Key Vault\n",
    "        for json_key, secret_name in secret_names.items():\n",
    "            secret_value = firebase_json[json_key]\n",
    "            client.set_secret(secret_name, secret_value)\n",
    "            logger.info(f\"Successfully set secret: {secret_name}\")\n",
    "\n",
    "    except Exception as e:\n",
    "        logger.exception(f\"Error setting secrets: {e}\")\n",
    "        raise\n",
    "\n",
    "# Run the function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "46017db1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:azure.identity._credentials.environment:No environment configuration found.\n",
      "INFO:azure.identity._credentials.managed_identity:ManagedIdentityCredential will use IMDS\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-TYPE?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '0'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': 'ffe571c4-3b03-11f0-abe6-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "No body was attached to the request\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): kvfactanalysis.vault.azure.net:443\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-TYPE?api-version=7.5 HTTP/1.1\" 401 97\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 401\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': 'ffe571c4-3b03-11f0-abe6-b0227adee090'\n",
      "    'x-ms-request-id': 'b6d87c96-1ac6-4a5d-993c-d1b9b107fc9f'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'WWW-Authenticate': 'Bearer authorization=\"https://login.microsoftonline.com/5c3f2b30-36c5-4d59-91ab-833ee8987361\", resource=\"https://vault.azure.net\"'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:01 GMT'\n",
      "    'Content-Length': '97'\n",
      "DEBUG:azure.identity._internal.decorators:EnvironmentCredential.get_token_info failed: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.\n",
      "Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\decorators.py\", line 23, in wrapper\n",
      "    token = fn(*args, **kwargs)\n",
      "            ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\environment.py\", line 181, in get_token_info\n",
      "    raise CredentialUnavailableError(message=message)\n",
      "azure.identity._exceptions.CredentialUnavailableError: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.\n",
      "Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=REDACTED&resource=REDACTED'\n",
      "Request method: 'GET'\n",
      "Request headers:\n",
      "    'User-Agent': 'azsdk-python-identity/1.23.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "No body was attached to the request\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254:80\n",
      "DEBUG:azure.identity._internal.msal_managed_identity_client:ImdsCredential.get_token_info failed: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\imds.py\", line 86, in _request_token\n",
      "    client.request_token(*scopes, connection_timeout=1, retry_total=0)\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\managed_identity_client.py\", line 145, in request_token\n",
      "    response = self._pipeline.run(request, retry_on_methods=[request.method], **kwargs)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 242, in run\n",
      "    return first_node.send(pipeline_request)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  [Previous line repeated 1 more time]\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\policies\\_retry.py\", line 566, in send\n",
      "    raise err\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\policies\\_retry.py\", line 544, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  [Previous line repeated 1 more time]\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 130, in send\n",
      "    self._sender.send(request.http_request, **request.context.options),\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\transport\\_requests_basic.py\", line 409, in send\n",
      "    raise error\n",
      "azure.core.exceptions.ServiceRequestError: (<urllib3.connection.HTTPConnection object at 0x0000028D70935050>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)')\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\msal_managed_identity_client.py\", line 163, in _get_token_base\n",
      "    token = self._request_token(*scopes, claims=claims, tenant_id=tenant_id, enable_cae=enable_cae, **kwargs)\n",
      "            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\imds.py\", line 96, in _request_token\n",
      "    raise CredentialUnavailableError(error_message) from ex\n",
      "azure.identity._exceptions.CredentialUnavailableError: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.\n",
      "DEBUG:azure.identity._internal.decorators:ManagedIdentityCredential.get_token_info failed: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\imds.py\", line 86, in _request_token\n",
      "    client.request_token(*scopes, connection_timeout=1, retry_total=0)\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\managed_identity_client.py\", line 145, in request_token\n",
      "    response = self._pipeline.run(request, retry_on_methods=[request.method], **kwargs)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 242, in run\n",
      "    return first_node.send(pipeline_request)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  [Previous line repeated 1 more time]\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\policies\\_retry.py\", line 566, in send\n",
      "    raise err\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\policies\\_retry.py\", line 544, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 98, in send\n",
      "    response = self.next.send(request)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  [Previous line repeated 1 more time]\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\_base.py\", line 130, in send\n",
      "    self._sender.send(request.http_request, **request.context.options),\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\core\\pipeline\\transport\\_requests_basic.py\", line 409, in send\n",
      "    raise error\n",
      "azure.core.exceptions.ServiceRequestError: (<urllib3.connection.HTTPConnection object at 0x0000028D70935050>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)')\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\decorators.py\", line 23, in wrapper\n",
      "    token = fn(*args, **kwargs)\n",
      "            ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\managed_identity.py\", line 212, in get_token_info\n",
      "    return cast(SupportsTokenInfo, self._credential).get_token_info(*scopes, options=options)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\msal_managed_identity_client.py\", line 143, in get_token_info\n",
      "    return self._get_token_base(*scopes, options=options, base_method_name=\"get_token_info\")\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\msal_managed_identity_client.py\", line 163, in _get_token_base\n",
      "    token = self._request_token(*scopes, claims=claims, tenant_id=tenant_id, enable_cae=enable_cae, **kwargs)\n",
      "            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\imds.py\", line 96, in _request_token\n",
      "    raise CredentialUnavailableError(error_message) from ex\n",
      "azure.identity._exceptions.CredentialUnavailableError: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.\n",
      "DEBUG:azure.identity._internal.decorators:SharedTokenCacheCredential.get_token_info failed: SharedTokenCacheCredential authentication unavailable. Multiple accounts\n",
      "were found in the cache. Use username and tenant id to disambiguate.\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\decorators.py\", line 23, in wrapper\n",
      "    token = fn(*args, **kwargs)\n",
      "            ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\shared_cache.py\", line 110, in get_token_info\n",
      "    return cast(SupportsTokenInfo, self._credential).get_token_info(*scopes, options=options)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\shared_cache.py\", line 156, in get_token_info\n",
      "    return self._get_token_base(*scopes, options=options, base_method_name=\"get_token_info\")\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_credentials\\shared_cache.py\", line 186, in _get_token_base\n",
      "    account = self._get_account(self._username, self._tenant_id, is_cae=is_cae)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\decorators.py\", line 75, in wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\ShiyamKrishnappa\\anaconda3\\envs\\fact_env_\\Lib\\site-packages\\azure\\identity\\_internal\\shared_token_cache.py\", line 227, in _get_account\n",
      "    raise CredentialUnavailableError(message=message)\n",
      "azure.identity._exceptions.CredentialUnavailableError: SharedTokenCacheCredential authentication unavailable. Multiple accounts\n",
      "were found in the cache. Use username and tenant id to disambiguate.\n",
      "DEBUG:azure.identity._credentials.azure_cli:Executing subprocess with the following arguments ['C:\\\\Program Files\\\\Microsoft SDKs\\\\Azure\\\\CLI2\\\\wbin\\\\az.CMD', 'account', 'get-access-token', '--output', 'json', '--resource', 'https://vault.azure.net', '--tenant', '5c3f2b30-36c5-4d59-91ab-833ee8987361']\n",
      "DEBUG:azure.identity._internal.decorators:AzureCliCredential.get_token_info succeeded\n",
      "DEBUG:azure.identity._internal.decorators:[Authenticated account] Client ID: 04b07795-8ddb-461a-bbee-02f9e1bf7b46. Tenant ID: 5c3f2b30-36c5-4d59-91ab-833ee8987361. User Principal Name: unavailableUpn. Object ID (user): df276354-470c-4556-8fa9-3996a67dde4d\n",
      "INFO:azure.identity._credentials.chained:DefaultAzureCredential acquired a token from AzureCliCredential\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-TYPE?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '28'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': 'ffe571c4-3b03-11f0-abe6-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-TYPE?api-version=7.5 HTTP/1.1\" 200 255\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': 'ffe571c4-3b03-11f0-abe6-b0227adee090'\n",
      "    'x-ms-request-id': '9ef04c7c-03c7-480c-9b1a-5530a7095781'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:03 GMT'\n",
      "    'Content-Length': '255'\n",
      "INFO:__main__:Successfully set secret: FB-TYPE\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-PROJETCID?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '26'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '020ec018-3b04-11f0-9b93-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-PROJETCID?api-version=7.5 HTTP/1.1\" 200 258\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '020ec018-3b04-11f0-9b93-b0227adee090'\n",
      "    'x-ms-request-id': '90a09134-68f1-4f1c-8c25-872d88eb6e86'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:03 GMT'\n",
      "    'Content-Length': '258'\n",
      "INFO:__main__:Successfully set secret: FB-PROJETCID\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-PRIVATEKEYID?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '53'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '02378410-3b04-11f0-a856-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-PRIVATEKEYID?api-version=7.5 HTTP/1.1\" 200 288\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '02378410-3b04-11f0-a856-b0227adee090'\n",
      "    'x-ms-request-id': '2d544aed-0d44-40db-a549-d45131e85bcf'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:04 GMT'\n",
      "    'Content-Length': '288'\n",
      "INFO:__main__:Successfully set secret: FB-PRIVATEKEYID\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-PRIVATEKEY?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '1689'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '026ea677-3b04-11f0-9def-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-PRIVATEKEY?api-version=7.5 HTTP/1.1\" 200 1922\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '026ea677-3b04-11f0-9def-b0227adee090'\n",
      "    'x-ms-request-id': '01644ab2-07f5-4db0-a4c3-7284eaf13262'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:04 GMT'\n",
      "    'Content-Length': '1922'\n",
      "INFO:__main__:Successfully set secret: FB-PRIVATEKEY\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-CLIENTEMAIL?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '74'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '029ba8bf-3b04-11f0-8136-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-CLIENTEMAIL?api-version=7.5 HTTP/1.1\" 200 308\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '029ba8bf-3b04-11f0-8136-b0227adee090'\n",
      "    'x-ms-request-id': '275d70d7-5337-4347-8e6c-851503703a53'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:04 GMT'\n",
      "    'Content-Length': '308'\n",
      "INFO:__main__:Successfully set secret: FB-CLIENTEMAIL\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-CLIENTID?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '34'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '02c926df-3b04-11f0-b927-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-CLIENTID?api-version=7.5 HTTP/1.1\" 200 265\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '02c926df-3b04-11f0-b927-b0227adee090'\n",
      "    'x-ms-request-id': '119626cc-4f05-47f5-b2c3-bf9aea29a7a0'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:05 GMT'\n",
      "    'Content-Length': '265'\n",
      "INFO:__main__:Successfully set secret: FB-CLIENTID\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-AUTHURI?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '54'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '02f69016-3b04-11f0-8842-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-AUTHURI?api-version=7.5 HTTP/1.1\" 200 284\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '02f69016-3b04-11f0-8842-b0227adee090'\n",
      "    'x-ms-request-id': '379ddcfd-2fe4-47f3-965a-becac0ecae63'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:05 GMT'\n",
      "    'Content-Length': '284'\n",
      "INFO:__main__:Successfully set secret: FB-AUTHURI\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-TOKENURI?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '48'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '0323d91b-3b04-11f0-b2e7-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-TOKENURI?api-version=7.5 HTTP/1.1\" 200 279\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '0323d91b-3b04-11f0-b2e7-b0227adee090'\n",
      "    'x-ms-request-id': 'f4fa33df-e39a-42d5-b38e-ea30ba66943f'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:05 GMT'\n",
      "    'Content-Length': '279'\n",
      "INFO:__main__:Successfully set secret: FB-TOKENURI\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-AUTHPROVIDERCERTURL?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '55'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '034d3c38-3b04-11f0-b302-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-AUTHPROVIDERCERTURL?api-version=7.5 HTTP/1.1\" 200 297\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '034d3c38-3b04-11f0-b302-b0227adee090'\n",
      "    'x-ms-request-id': 'f04d9a65-f7c6-4a28-9cc2-d8e398a6ad89'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:06 GMT'\n",
      "    'Content-Length': '297'\n",
      "INFO:__main__:Successfully set secret: FB-AUTHPROVIDERCERTURL\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-CLIENTURL?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '126'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '0377a395-3b04-11f0-a5cd-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-CLIENTURL?api-version=7.5 HTTP/1.1\" 200 358\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '0377a395-3b04-11f0-a5cd-b0227adee090'\n",
      "    'x-ms-request-id': '1d30adbb-b951-42ca-826c-735a615494c1'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:06 GMT'\n",
      "    'Content-Length': '358'\n",
      "INFO:__main__:Successfully set secret: FB-CLIENTURL\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://kvfactanalysis.vault.azure.net/secrets/FB-UNIVERSEDOMAIN?api-version=REDACTED'\n",
      "Request method: 'PUT'\n",
      "Request headers:\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Length': '27'\n",
      "    'Accept': 'application/json'\n",
      "    'x-ms-client-request-id': '03a55994-3b04-11f0-9eda-b0227adee090'\n",
      "    'User-Agent': 'azsdk-python-keyvault-secrets/4.9.0 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'Authorization': 'REDACTED'\n",
      "A body is sent with the request\n",
      "DEBUG:urllib3.connectionpool:https://kvfactanalysis.vault.azure.net:443 \"PUT /secrets/FB-UNIVERSEDOMAIN?api-version=7.5 HTTP/1.1\" 200 264\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 200\n",
      "Response headers:\n",
      "    'Cache-Control': 'no-cache'\n",
      "    'Pragma': 'no-cache'\n",
      "    'Content-Type': 'application/json; charset=utf-8'\n",
      "    'Expires': '-1'\n",
      "    'x-ms-keyvault-region': 'eastus'\n",
      "    'x-ms-client-request-id': '03a55994-3b04-11f0-9eda-b0227adee090'\n",
      "    'x-ms-request-id': '7eea706f-7676-4bdb-b2bc-8e1603eb2e2a'\n",
      "    'x-ms-keyvault-service-version': '1.9.2421.1'\n",
      "    'x-ms-keyvault-network-info': 'conn_type=Ipv4;addr=182.74.29.37;act_addr_fam=InterNetwork;'\n",
      "    'x-ms-keyvault-rbac-assignment-id': 'REDACTED'\n",
      "    'X-Content-Type-Options': 'REDACTED'\n",
      "    'Strict-Transport-Security': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:08:06 GMT'\n",
      "    'Content-Length': '264'\n",
      "INFO:__main__:Successfully set secret: FB-UNIVERSEDOMAIN\n"
     ]
    }
   ],
   "source": [
    "set_keyvault_secrets()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e88fdfc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "from azure.storage.blob import BlobServiceClient\n",
    "import json\n",
    "from firebase_admin import credentials, initialize_app\n",
    "import logging\n",
    "import os\n",
    "from io import BytesIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a0887648",
   "metadata": {},
   "outputs": [],
   "source": [
    "logging.basicConfig(level=logging.DEBUG) \n",
    "logger = logging.getLogger(__name__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1dd706ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "AZURE_STORAGE_ACCOUNT_URL = \"https://safactcheck.blob.core.windows.net\"\n",
    "AZURE_STORAGE_KEY = 'imoBXY2yicpv5YiA4JBxc3ckudu/FGc4yxaifh6jF9wLudv/v26tuJqY/gNdAMKprRcAyWtxABIZ+ASteADMBA=='\n",
    "container_name = \"firebase-private-key\"\n",
    "blob_name = \"fact-check257-firebase-adminsdk-fbsvc-5800407e8b.json\"\n",
    "local_file_name = \"firebase_admin_creds.json\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57a51ce1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def download_pdf(blob_name: str, container_name: str):\n",
    "    try:\n",
    "        blob_service_client = BlobServiceClient(\n",
    "            account_url=AZURE_STORAGE_ACCOUNT_URL, \n",
    "            credential=AZURE_STORAGE_KEY\n",
    "        )\n",
    "        container_client = blob_service_client.get_container_client(container_name)\n",
    "        blob_client = container_client.get_blob_client(blob_name)\n",
    "\n",
    "        stream = BytesIO()\n",
    "        blob_client.download_blob().readinto(stream)\n",
    "        stream.seek(0) \n",
    "        return stream\n",
    "\n",
    "    except Exception as e:\n",
    "        raise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "371ec52f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://safactcheck.blob.core.windows.net/firebase-private-key/fact-check257-firebase-adminsdk-fbsvc-5800407e8b.json'\n",
      "Request method: 'GET'\n",
      "Request headers:\n",
      "    'x-ms-range': 'REDACTED'\n",
      "    'x-ms-version': 'REDACTED'\n",
      "    'Accept': 'application/xml'\n",
      "    'User-Agent': 'azsdk-python-storage-blob/12.25.1 Python/3.11.11 (Windows-10-10.0.19045-SP0)'\n",
      "    'x-ms-date': 'REDACTED'\n",
      "    'x-ms-client-request-id': 'a354f92a-3b06-11f0-bdde-b0227adee090'\n",
      "    'Authorization': 'REDACTED'\n",
      "No body was attached to the request\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): safactcheck.blob.core.windows.net:443\n",
      "DEBUG:urllib3.connectionpool:https://safactcheck.blob.core.windows.net:443 \"GET /firebase-private-key/fact-check257-firebase-adminsdk-fbsvc-5800407e8b.json HTTP/1.1\" 206 2379\n",
      "INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 206\n",
      "Response headers:\n",
      "    'Content-Length': '2379'\n",
      "    'Content-Type': 'application/json'\n",
      "    'Content-Range': 'REDACTED'\n",
      "    'Last-Modified': 'Tue, 27 May 2025 14:11:53 GMT'\n",
      "    'Accept-Ranges': 'REDACTED'\n",
      "    'ETag': '\"0x8DD9D286E91E896\"'\n",
      "    'Server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0'\n",
      "    'x-ms-request-id': '0989c92c-d01e-0057-7613-cfa0fd000000'\n",
      "    'x-ms-client-request-id': 'a354f92a-3b06-11f0-bdde-b0227adee090'\n",
      "    'x-ms-version': 'REDACTED'\n",
      "    'x-ms-creation-time': 'REDACTED'\n",
      "    'x-ms-blob-content-md5': 'REDACTED'\n",
      "    'x-ms-lease-status': 'REDACTED'\n",
      "    'x-ms-lease-state': 'REDACTED'\n",
      "    'x-ms-blob-type': 'REDACTED'\n",
      "    'x-ms-server-encrypted': 'REDACTED'\n",
      "    'Date': 'Tue, 27 May 2025 14:26:54 GMT'\n"
     ]
    }
   ],
   "source": [
    "data = download_pdf(blob_name, container_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "3c9fbe73",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "  \"type\": \"service_account\",\n",
      "  \"project_id\": \"fact-check257\",\n",
      "  \"private_key_id\": \"5800407e8b1f6d04368c2e935d20e2659dd9e4d0\",\n",
      "  \"private_key\": \"-----BEGIN PRIVATE KEY-----\\nMIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCkqbvKaTaSNCqL\\nKm8EbLN1jedYSojQP+j5gA0sqwSL7mF9fr4A5CrktbQinvubpwS6lJL/e82pWbNQ\\nHVPexycj4zei0hR/0yOZrOGGH6OpPT8bTq+vq730bSKSDNHOUzHFgyAVRVQo+kYK\\nCLmtpHuXomw3yEDAQRKPxJdLf7kiZQS9pwAOGpjkD3P+teVp9ml/3agVWKtNcp3A\\nMM/BaUrRfCi3rMOph4PdrG6XVAftyP9geBZUDZ0IgBe213UQ4MONLu3b/+JrFc0c\\n6kbZdY6ysNyUjet2N+F6UFwViQE9IcDISOYtJn8LjuTQeAwTXNuLF4OlHHvV9vP5\\nDxkRAgQpAgMBAAECggEAONc6swmUw+1sDjQstItNe7Pre0fnaHPXWWFBqvyyChfU\\n4XxB3I7m26J5t0XINxf1gu7Mu5KiEOadBnfJPuricPn7d6IrX2CkNmU+SI1dcoto\\n4GVOMkcHwHt35pC0UBPZ54N3LMCM58RVLxuJ9Q4Fd/eigbMH3xXXcXd8fpxtbRiB\\nXSuuk9Vouhf7lnwQLWyO663QrvHxTylTtmWELScyIBxvDskSsC8f18IVUJAT5pMO\\nRNmPUMII8gTsCZa6gU4HoaMGk2AHK0sLiLK2xRHOSOS8KHdape+UIQ2Cx79RK9me\\nvzi18rp/XIGEPKuLx46EgmQuZG/Sm8jvZjl4hBQgfQKBgQDl6vYz+sCfZI7jRUgt\\nSEvjnbuzxeKXVUYGlj0rXh3VWzsXSNCFYLzESpOeEOtRiBtBuHAs9CyuoYDxBNG3\\nKhMgrKaYyiCOi4Kt2HANHSiqDFl1eXkYySz+CuoWjNijK4iUy7dD78exeqSa4A0e\\n/3tG7YOB+NP+cnR+yAYotY32uwKBgQC3V7VtoIpSOHEIPN3uhYg1j212G093GH8e\\nb/6ifZPSK3YAhktK15QO5oev5NjkrAHxwYQreoBDQoEdI5SPFlXbpDj7kVP6LgOj\\nZSHjX/qFRXRNkyYUFCOogYsBRf3nsOf6/Y87nFb7kLOaFnEN4Z+y0ZnWMLMYtyaU\\n7k5xtu9sawKBgHimpxI52NEL/OuIlMqmcKR+RlydUD3n0rpsJqdWWUN3/3dFj58e\\nV5COLg65hDMhwGPpfHwqmjP9rmNMCx0aKDVyRG5TS2hiD25Y8NwA3DD4EA3Uc9as\\n4z5gzdyNhrWBh8fFa2NsNZeN60kWn+wPV0quIt+WivCwcS+CmbDNLTU3AoGAGUXj\\n4orAxni1TBdn2Ftb3VY6Fo/JMneT/lQd27RHzi7jg0r/0f9Jj4xDpAF9zHfRhMvS\\nNlQanirqtWOUNO01RKOsnzD3TINPSRTFyRwP4pMiZxcjppzoFXBWAwTs7SU7tSOC\\nvQreCqt1toXla65wXyv5cscfyQZmMR1wfshi1KUCgYApqRkuP8dx4g/rgSRDOKbw\\n+kBknUkxVgcStrSA8U+DxMiGLgpELPSui9fORb4UNloUBFalQ7VY3EpnlISoS+l/\\n10eZVkmEUIOFuU0gcsaNCibzMWsNh4pQqsAzujz609BevvBxEl6+1Gc8jG48h8fB\\nq3/rwMMdXcw+7PhELF0/RA==\\n-----END PRIVATE KEY-----\\n\",\n",
      "  \"client_email\": \"firebase-adminsdk-fbsvc@fact-check257.iam.gserviceaccount.com\",\n",
      "  \"client_id\": \"101709437156448516768\",\n",
      "  \"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\n",
      "  \"token_uri\": \"https://oauth2.googleapis.com/token\",\n",
      "  \"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\n",
      "  \"client_x509_cert_url\": \"https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-fbsvc%40fact-check257.iam.gserviceaccount.com\",\n",
      "  \"universe_domain\": \"googleapis.com\"\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "# Convert BytesIO data to JSON string\n",
    "data.seek(0)\n",
    "json_content = json.load(data)\n",
    "print(json.dumps(json_content, indent=2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "cd230c27",
   "metadata": {},
   "outputs": [],
   "source": [
    "from firebase_admin import credentials, initialize_app\n",
    "\n",
    "\n",
    "cred = credentials.Certificate(cert=json_content)\n",
    "firebase_app = initialize_app(cred)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e018fbd9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<firebase_admin.credentials.Certificate at 0x28d7302d850>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "firebase_app.credential."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d87edc77",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "fact_env_",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
